### PR TITLE
cluster: use fqdn as url

### PR DESCRIPTION
**Pull request by xiang90**
fix https://github.com/coreos/etcd-operator/issues/901

### CODE REVIEW

**Comment by xiang90** [comment]
@colhom 

probably we should just remove the member peer service thing and use https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/ instead?

**Comment by colhom** [comment]
@xiang90 I would agree- the peer FQDN only lives as long as the pod, so there is no need technically for the service abstraction for the peer interface.

**Comment by xiang90** [comment]
@hongchaodeng talked with @colhom. we will first get this simple pr merged, so we use domain names. then we switch to use pod sub domain name to get rid of the service domain name.

File: pkg/util/etcdutil/member.go [hunk 1]
```diff
@@ -140,6 +148,20 @@ func GetCounterFromMemberName(name string) (int, error) {
 	return c, nil
 }
 
+func MemberNameFromPeerURL(pu string) (string, error) {
+	// url.Parse has very loose validation. We do our own validation.
+	if !validPeerURL.MatchString(pu) {
```
Comment (hongchaodeng) [inline pkg/util/etcdutil/member.go]: Also move `validPeerURL` to this file?

Comment (xiang90) [inline pkg/util/etcdutil/member.go]: fixed.

**Pull request merged**